src/java.base/share/classes/java/lang/BaseVirtualThread.java

```diff
@@ -63,5 +68,13 @@ sealed abstract class BaseVirtualThread extends Thread
      * Makes available the parking permit to the given this virtual thread.
      */
     abstract void unpark();
+
+    @Hidden
+    @ForceInline
+    protected void runWith(Object bindings, Runnable op) {
+        ensureMaterializedForStackWalk(bindings);
+        op.run();
+        Reference.reachabilityFence(bindings);
+    }
```

Comment:
BaseVirtualThread <: Thread.  Would it be simpler, and avoid duplicate, to just change Thread.runWith to be a final method with default access. That would be accessibility to VirtualThread and BoundVirtualThread without duplication.

Reply:
Sure, that makes sense.